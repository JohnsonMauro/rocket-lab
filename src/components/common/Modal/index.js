import React, { useState, useEffect, useCallback } from 'react';

import {
  ModalStyles,
  ModalContent,
  ModalHeader,
  ModalHeaderTitle,
  ModalHeaderSubTitle,
  ModalBody,
} from './styles';

const Modal = ({ children, isInitiallyOpened, onOpening, subTitle, title }) => {
  const [isOpened, setOpeningStatus] = useState(false);

  useEffect(() => {
    if (isInitiallyOpened) {
      setOpeningStatus(true);
    }
  }, [isInitiallyOpened]);

  useEffect(() => {
    if (isOpened) {
      onOpening();
      setOpeningStatus(true);
    }
  }, [isOpened]);

  const handleClosingClick = useCallback(() => {
    onClosing();
    setOpeningStatus(false);
  }, []);

  const handleDismiss = useCallback((e) => {
    if (e.target.id === 'dismiss-modal') {
      handleClosingClick();
    }
  }, []);

  if (!isOpened) {
    return null;
  }

  return (
    <ModalStyles aria-modal={isOpened} id="dismiss-modal" onClick={handleDismiss} role="dialog">
      <ModalContent>
        <ModalHeader>
          <ModalHeaderTitle>{title}</ModalHeaderTitle>
          <ModalHeaderSubTitle>{subTitle}</ModalHeaderSubTitle>
        </ModalHeader>
        <ModalBody>{children}</ModalBody>
      </ModalContent>
    </ModalStyles>
  );
};

Modal.defaultProps = {
  isInitiallyOpened: false,
  onOpening: () => {},
  onClosing: () => {},
};

export default Modal;
