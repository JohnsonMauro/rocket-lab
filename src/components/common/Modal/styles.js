import styled from '@emotion/styled';

const ModalStyles = styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.modalBackGroung};
  bottom: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 99;
`;

const ModalContent = styled.div`
  background-color: ${({ theme }) => theme.colors.white};
  border: 0.2px solid ${({ theme }) => theme.colors.gainsboro};
  border-radius: 1.8px;
  box-shadow: 0px 1px 20px rgba(0, 0, 0, 0.2);
  min-height: 903px;
  overflow-y: auto;
  padding: 0 50px;
  position: relative;
  width: 936px;
  z-index: 99;
  -webkit-overflow-scrolling: touch;

  section {
    padding: 0;
  }

  @media (max-width: 992px) {
    padding: 0 25px;
    width: 85%;
  }

  @media (max-width: 576px) {
    box-shadow: none;
    height: 100%;
    min-height: unset;
    width: 100%;
  }
`;

const ModalHeader = styled.div`
  height: 117px;
`;

const ModalHeaderTitle = styled.h1`
  color: ${({ theme }) => theme.colors.blackRussian};
  display: inline-block;
  font-size: 25px;
  font-weight: 400;
  margin: 50px 15px 13px 0;
`;

const ModalHeaderImg = styled.img`
  height: 16px;
  width: 22px;
`;

const ModalHeaderClose = styled.img`
  cursor: pointer;
  height: 16px;
  position: absolute;
  right: 37px;
  top: 30px;
  width: 16px;

  @media (max-width: 576px) {
    right: 23px;
    top: 19px;
  }
`;

const ModalHeaderSubTitle = styled.h4`
  color: ${({ theme }) => theme.colors.blackRussian};
  font-weight: 400;
  margin: 0;
  opacity: 0.5;
`;

const ModalBody = styled.div``;

export {
  ModalStyles,
  ModalContent,
  ModalHeader,
  ModalHeaderTitle,
  ModalHeaderImg,
  ModalHeaderClose,
  ModalHeaderSubTitle,
  ModalBody,
};
