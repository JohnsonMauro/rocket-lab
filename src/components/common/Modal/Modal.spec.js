import React from 'react';
import { render } from '@testing-library/react';

import GlobalsContainer from '@test/GlobalsContainer';
import Modal from './index';
import Check from '@svg/check.svg';

const isOpen = false;

describe('<Modal />', () => {
  it('Renders the Modal component when is close', () => {
    const { container } = render(
      <Modal icon={Check} subTitle="Title sample for modal" title="Sample" initialOpened={isOpen}>
        Sample
      </Modal>,
      {
        wrapper: GlobalsContainer,
      }
    );
    expect(container).toMatchSnapshot();
  });

  it('Renders the Modal component when is open', () => {
    const { container } = render(
      <Modal icon={Check} subTitle="Title sample for modal" title="Sample" initialOpened={!isOpen}>
        Sample
      </Modal>,
      {
        wrapper: GlobalsContainer,
      }
    );
    expect(container).toMatchSnapshot();
  });
});
