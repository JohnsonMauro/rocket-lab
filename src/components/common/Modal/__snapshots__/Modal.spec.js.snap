// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> Renders the Modal component when is close 1`] = `<div />`;

exports[`<Modal /> Renders the Modal component when is open 1`] = `
.emotion-14 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(0,0,0,0.5);
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 99;
}

.emotion-12 {
  background-color: #ffffff;
  border: 0.2px solid #E1E1E1;
  border-radius: 1.8px;
  box-shadow: 0px 1px 20px rgba(0,0,0,0.2);
  min-height: 903px;
  overflow-y: auto;
  padding: 0 50px;
  position: relative;
  width: 936px;
  z-index: 99;
  -webkit-overflow-scrolling: touch;
}

.emotion-12 section {
  padding: 0;
}

@media (max-width:992px) {
  .emotion-12 {
    padding: 0 25px;
    width: 85%;
  }
}

@media (max-width:576px) {
  .emotion-12 {
    box-shadow: none;
    height: 100%;
    min-height: unset;
    width: 100%;
  }
}

.emotion-8 {
  height: 117px;
}

.emotion-0 {
  cursor: pointer;
  height: 16px;
  position: absolute;
  right: 37px;
  top: 30px;
  width: 16px;
}

@media (max-width:576px) {
  .emotion-0 {
    right: 23px;
    top: 19px;
  }
}

.emotion-2 {
  color: #050925;
  display: inline-block;
  font-size: 25px;
  font-weight: 400;
  margin: 50px 15px 13px 0;
}

.emotion-4 {
  height: 16px;
  width: 22px;
}

.emotion-6 {
  color: #050925;
  font-weight: 400;
  margin: 0;
  opacity: 0.5;
}

<div>
  <div
    aria-modal="true"
    class="emotion-14 emotion-15"
    data-testid="modal-component"
    id="dismiss-modal"
    role="dialog"
  >
    <div
      class="emotion-12 emotion-13"
    >
      <div
        class="emotion-8 emotion-9"
      >
        <img
          class="emotion-0 emotion-1"
          data-testid="modal-close-button"
          src="mocked-file"
        />
        <h1
          class="emotion-2 emotion-3"
        >
          Sample
        </h1>
        <img
          class="emotion-4 emotion-5"
          src="mocked-file"
        />
        <h4
          class="emotion-6 emotion-7"
        >
          Title sample for modal
        </h4>
      </div>
      <div
        class="emotion-10 emotion-11"
      >
        Sample
      </div>
    </div>
  </div>
</div>
`;
